### PR TITLE
Add CFDate, CFTimeZone and make them work with chrono

### PR DESCRIPTION
**Pull request by ratake**
This change is [Reviewable](https://reviewable.io/reviews/servo/core-foundation-rs/126)

### CODE REVIEW

File: core-foundation/src/date.rs [hunk 1]
```diff
@@ -9,11 +9,17 @@
 
 //! Core Foundation date objects.
 
+#[cfg(feature = "with-chrono")]
+extern crate chrono;
```
Comment (jdm) [inline core-foundation/src/date.rs]: Let's move this into lib.rs instead.

File: core-foundation/src/timezone.rs [hunk 2]
```diff
@@ -9,12 +9,18 @@
 
 //! Core Foundation time zone objects.
 
+#[cfg(feature = "with-chrono")]
+extern crate chrono;
```
Comment (jdm) [inline core-foundation/src/timezone.rs]: This can be removed.

File: core-foundation/src/timezone.rs [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+#[cfg(test)]
+mod test {
+    use super::CFTimeZone;
+
+    #[test]
+    fn timezone_comparison() {
+        let system = CFTimeZone::system();
+        let default = CFTimeZone::default();
+        assert!(system == default);
```
Comment (jdm) [inline core-foundation/src/timezone.rs]: assert_eq

File: core-foundation/src/timezone.rs [hunk 4]
```diff
@@ -59,16 +65,39 @@ impl CFTimeZone {
...
         assert!(system == default);
     }
+
+    #[test]
+    #[cfg(feature = "with-chrono")]
+    fn timezone_chrono_conversion() {
+        let offset = FixedOffset::west(28800);
+        let tz = CFTimeZone::from_offset(offset);
+        let converted = tz.offset_at_date(NaiveDateTime::from_timestamp(0, 0));
+        assert!(offset == converted);
```
Comment (jdm) [inline core-foundation/src/timezone.rs]: assert_eq

File: core-foundation/src/date.rs [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+        assert!(now.cmp(&past) == Ordering::Greater);
+        assert!(now.cmp(&now) == Ordering::Equal);
+        assert!(past.cmp(&now) == Ordering::Less);
+    }
+
+    #[test]
+    fn date_equality() {
+        let now = CFDate::now();
+        let same_time = CFDate::new(now.abs_time());
+        assert!(now == same_time);
```
Comment (jdm) [inline core-foundation/src/date.rs]: assert_eq in all these tests.

**Comment by ratake** [comment]
I had to implement `Debug` for `CFDate`/`CFTimeZone` to use `assert_eq`, so I implemented `Debug` for all types that don't already have it implemented (and now use `assert_eq`)

**Comment by jdm** [comment]
@bors-servo: r+
Thanks!

**Pull request merged**